### PR TITLE
[FLINK-4303] [CEP] Add CEP examples

### PR DESCRIPTION
**Pull request by kisimple**
## What is the purpose of the change

Currently neither CEP Java nor CEP Scala contain a runnable example. This PR fixes the problem by adding a `flink-examples-cep` module. The change is based on #2937

## Brief change log

  - Add a `flink-examples-cep` module
  - Add a `TemperatureMonitoring.java` example
  - Add a `TemperatureMonitoring.scala` example

## Verifying this change

  - Build the project and run the example by the following commands:
`./bin/flink run -c org.apache.flink.cep.examples.java.monitoring.TemperatureMonitoring ./examples/cep/flink-examples-cep-with-dependencies.jar`
and
`./bin/flink run -c org.apache.flink.cep.examples.scala.monitoring.TemperatureMonitoring ./examples/cep/flink-examples-cep-with-dependencies.jar`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by medcv** [comment]
@kisimple Recently I started to work on same example. I ran your code on my local and it did work perfectly. I could see some checkstyle errors that I think travis might pick them up. 

I have a suggestion for the sake of example scenario. 
For the `alerts` event you using `LocalTime` which indicate when `alerts` occurred, it would be useful also add `warnings` timestamp to `alerts` event in order to show when racks temperature passed the threshold, something like this on [line](https://github.com/kisimple/flink/blob/5cd3a374b84b2a7aaedb4c4184caded073e19295/flink-examples/flink-examples-cep/src/main/java/org/apache/flink/cep/examples/java/monitoring/TemperatureMonitoring.java#L133)
 
`out.collect(new TemperatureAlert(first.getRackID(), second.getDatetime()));
`

of course it needs to update the `TemperatureAlert` model also.

File: flink-examples/flink-examples-cep/src/main/java/org/apache/flink/cep/examples/java/monitoring/TemperatureMonitoring.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+			.<TemperatureWarning>begin("first")
+			.next("second")
+			.within(Time.seconds(20));
+
+		// Create a pattern stream from our alert pattern
+		PatternStream<TemperatureWarning> alertPatternStream = CEP.pattern(
+			warnings.keyBy("rackID"),
+			alertPattern);
+
+		// Generate a temperature alert iff the second temperature warning's average temperature
```
Comment (medcv) [inline flink-examples/flink-examples-cep/src/main/java/org/apache/flink/cep/examples/java/monitoring/TemperatureMonitoring.java]: type `iff`

File: flink-examples/flink-examples-cep/src/main/scala/org/apache/flink/cep/examples/scala/monitoring/TemperatureMonitoring.scala [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    // appearing within a time interval of 20 seconds
+    val alertPattern = Pattern
+      .begin[TemperatureWarning]("first")
+      .next("second")
+      .within(Time.seconds(20))
+
+    // Create a pattern stream from our alert pattern
+    val alertPatternStream = CEP.pattern(warnings.keyBy(_.rackID), alertPattern)
+
+    // Generate a temperature alert iff the second temperature warning's average temperature
```
Comment (medcv) [inline flink-examples/flink-examples-cep/src/main/scala/org/apache/flink/cep/examples/scala/monitoring/TemperatureMonitoring.scala]: type `iff`

File: flink-examples/flink-examples-cep/src/main/java/org/apache/flink/cep/examples/java/monitoring/TemperatureMonitoring.java (hunk 1 above)
Comment (kisimple) [reply flink-examples/flink-examples-cep/src/main/java/org/apache/flink/cep/examples/java/monitoring/TemperatureMonitoring.java]: Not a typo, just an abbreviation for `if and only if` :)

**Comment by kisimple** [comment]
cc @twalthr

File: flink-examples/flink-examples-cep/src/main/java/org/apache/flink/cep/examples/java/monitoring/TemperatureMonitoring.java (hunk 1 above)
Comment (medcv) [reply flink-examples/flink-examples-cep/src/main/java/org/apache/flink/cep/examples/java/monitoring/TemperatureMonitoring.java]: lol, didn't know lol

**Comment by medcv** [comment]
+1
@kisimple Thank you for update!

**Comment by kisimple** [comment]
cc @dawidwys

**Comment by tillrohrmann** [comment]
Closing because the PR is outdated.

**Pull request closed without merging**